packages/components/src/toolbar/toolbar/index.tsx

```diff
@@ -19,23 +19,30 @@ import type { ToolbarProps } from './types';
 import type { WordPressComponentProps } from '../../context';
 import { ContextSystemProvider } from '../../context';
 
-const CONTEXT_SYSTEM_VALUE = {
-	DropdownMenu: {
-		variant: 'toolbar',
-	},
-	Dropdown: {
-		variant: 'toolbar',
-	},
-};
-
 function UnforwardedToolbar(
 	{
 		className,
 		label,
+		variant,
 		...props
 	}: WordPressComponentProps< ToolbarProps, 'div', false >,
 	ref: ForwardedRef< any >
 ) {
+	const CONTEXT_SYSTEM_VALUE = useMemo( () => {
+		if ( variant !== undefined ) {
+			return {};
+		}
+
+		return {
+			DropdownMenu: {
+				variant: 'toolbar',
+			},
+			Dropdown: {
+				variant: 'toolbar',
+			},
+		};
+	}, [ variant ] );
```

Comment:
`CONTEXT_SYSTEM_VALUE` will be recalculated on every value, regardless what it is. At the same time, we only care about its declaration if it changes from `undefined` to any other value, or the other way around. 

So, to micro-optimize, I'd recommend declaring that as a value above the `useMemo` and using that instead of `variant`:

```suggestion
	const isVariantDefined = variant !== undefined;
	const CONTEXT_SYSTEM_VALUE = useMemo( () => {
		if ( isVariantDefined ) {
			return {};
		}

		return {
			DropdownMenu: {
				variant: 'toolbar',
			},
			Dropdown: {
				variant: 'toolbar',
			},
		};
	}, [ isVariantDefined ] );
```

Reply:
Thanks for sharing! I'm unfamiliar with `useMemo` and implemented it due to a `console.warn`. So, I'm unsure if I fully understand the difference. 🤔 If you have time to share more or if you know of any examples I could look into to understand further, I'd appreciate it!

Reply:
The code inside `useMemo` gets executed only when any of its dependencies change — it is therefore useful when we want to only re-compute something when absolutely necessary, and otherwise use the result from past computations. This is particularly useful for a bunch of reasons:
- it's great when we need to compute something particularly complex / slow to calculate (not this case)
- since React component re-render any time one of their props change, by memorizing a value we can try to reduce that number of rerenders. In fact, without `useMemo` we would be technically creating a new, different object every time

The [react docs](https://react.dev/reference/react/useMemo) are a good start if you're interested in learning more

Reply:
What @ciampo said 👍 

In terms of how often it recomputes, `useMemo()` is very similar to `useEffect` or `useCallback` - it will be recomputed only if one of the dependencies change, so we always strive to keep the dependencies as minimal and as static as possible. 

Let's consider an example. If `variant` is a dependency, then the `useMemo()` will be recompute every time `variant` changes, for example, if `variant` was `A` and gets changed to `B`, or if it was `B` and was changed to `undefined`. However, if we declare `const isVariantDefined = variant !== undefined;` and then use `isVariantDefined` as a dependency, `useMemo` will recompute only if `isVariantDefined` changes from `undefined` to anything else, or from anything else to `undefined`. That means, if `variant` changes from `A` to `B`, `isVariantDefined` won't change, therefore `useMemo()` won't recompute.

That being said, it's a micro-optimization, and likely won't have a measurable impact, unless we change the `variant` prop often for some reason.